internal/service/cluster/model_cluster.go

```diff
@@ -272,6 +276,14 @@ func expandProcessArgs(d *schema.ResourceData, p map[string]any) *matlas.Process
 		}
 	}
 
+	if _, ok := d.GetOkExists("advanced_configuration.0.change_stream_options_pre_and_post_images_expire_after_seconds"); ok {
+		if changeStreamOptionsSeconds := cast.ToInt64(p["change_stream_options_pre_and_post_images_expire_after_seconds"]); changeStreamOptionsSeconds != 0 {
```

Comment:
q: why do we need to check value is != 0 in this case? If value is not defined by users the attribute will have value -1 right?

Reply:
As per the documentation, valid values are integer values greater than zero or -1. I used the documentation from the API(which is hidden currently, so not showed in the API docs) but available in the code

Reply:
Given its an invalid value, should we instead intentional include in the request so that an error is returned? With current implementation not sure how the user experience would be, likely an non empty plan after the apply.

Reply:
Makes sense, I have changed it so that 0 value gets sent in the request letting the API fail